tests/e2e/pageobjects/ide/Terminal.ts

```diff
@@ -134,6 +134,24 @@ export class Terminal {
         }, timeout);
     }
 
+    public async waitIconSuccess(taskName: string, timeout: number) {
+        const terminalTabLocator: By = By.css(`${this.getTerminalTabCssLocator(taskName)} div.p-TabBar-tabIcon`);
+        await this.driverHelper.waitVisibility(terminalTabLocator, TimeoutConstants.TS_SELENIUM_TERMINAL_DEFAULT_TIMEOUT);
+        let terminalClass = await this.driverHelper.waitAndGetElementAttribute(terminalTabLocator, 'class');
+
+        await this.driverHelper.getDriver().wait(async () => {
+            terminalClass = await this.driverHelper.waitAndGetElementAttribute(terminalTabLocator, 'class');
+            if (terminalClass.includes('fa-check')) { // css for tick icon
+                return true;
+            }
+            if (terminalClass.includes('fa-times-circle')) { // css for failed icon
+                Logger.error('The error occured when waiting for task "' + taskName + '" to succeed.');
+                throw new Error('The error occured when waiting for task "' + taskName + '" to succeed.');
+            }
+            return false;
+        }, timeout);
```

Comment:
If you want, you can also specify error message for cases, when the `wait` times out for easier understanding of the logs if this case happens. Something like:
```suggestion
        }, timeout, "Timed out waiting for task " + taskName + " to succeed.");
        
```

Reply:
Good point, added.